### PR TITLE
Add retries to the nginx service starts to resolve restore issues

### PR DESCRIPTION
**Pull request by jashaik**
Signed-off-by: jan shahid shaik <jashaik@progress.com>

### Description

after applying backup nginx service is not getting started(runsv)

### Issues Resolved

This is resolved by adding retry mechanism.

### Check List

- [ ] New functionality includes tests
- [ ] All buildkite tests pass
- [ ] Full omnibus build and tests in buildkite pass
- [ ] All commits have been signed-off for the Developer Certificate of Origin. See <https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco>
- [ ] PR title is a worthy inclusion in the CHANGELOG

### CODE REVIEW

**Comment by jashaik** [comment]
Pipeline Information
adhoc - https://buildkite.com/chef/chef-chef-server-master-omnibus-adhoc/builds/2741#f7cdc0e1-2c05-4bde-9181-bc4ecdb0a761

File: omnibus/files/private-chef-cookbooks/private-chef/recipes/nginx.rb [hunk 1]
```diff
@@ -243,7 +243,11 @@
 end
 
 # Set up runit service for nginx component
-component_runit_service 'nginx'
+component_runit_service 'nginx' do
+  action [:enable, :start]
```
Comment (PrajaktaPurohit) [inline omnibus/files/private-chef-cookbooks/private-chef/recipes/nginx.rb:247]: are both the enable and start actions required here?

File: omnibus/files/private-chef-cookbooks/private-chef/recipes/nginx.rb [hunk 2]
```diff
@@ -243,7 +243,11 @@
 end
 
 # Set up runit service for nginx component
-component_runit_service 'nginx'
+component_runit_service 'nginx' do
```
Comment (PrajaktaPurohit) [inline omnibus/files/private-chef-cookbooks/private-chef/recipes/nginx.rb:246]: could you please add a comment about why the retries are necessary here?

**Comment by PrajaktaPurohit** [comment]
Looks like the build versions in the adhoc and umbrella branches do not match, do one or the other link just need to be updated?

Is there a scenario in umbrella testing this usecase?

File: omnibus/files/private-chef-cookbooks/private-chef/recipes/nginx.rb (hunk 1 above)
Comment (jashaik) [reply omnibus/files/private-chef-cookbooks/private-chef/recipes/nginx.rb:247]: Yes Prajakta, start is not working if i do multiple restores due to some missing directories. enable and start are solving these issues

**Comment by jashaik** [comment]
> Looks like the build versions in the adhoc and umbrella branches do not match, do one or the other link just need to be updated?
> 
> Is there a scenario in umbrella testing this usecase?

Currently we don't have any scenario in the umbrella but There was some work done on this by E2E team earlier.

**Review by PrajaktaPurohit — APPROVED** [review]

**Comment by tas50** [comment]
https://buildkite.com/chef/chef-chef-server-master-omnibus-adhoc/builds/2780#b1e1ac8f-fdb3-49e4-b8fb-28774ca210b2

**Pull request merged**